File: remote/src/main/java/com/kelsos/mbrc/events/ChangeWebSocketStatusEvent.java
package com.kelsos.mbrc.events;

import android.support.annotation.IntDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class ChangeWebSocketStatusEvent {
  public static final int CONNECT = 1;
  public static final int DISCONNECT = 2;

  @Action private int action;

  private ChangeWebSocketStatusEvent() {
    //no instance
  }

  public static ChangeWebSocketStatusEvent newInstance(@Action int action) {
    ChangeWebSocketStatusEvent event = new ChangeWebSocketStatusEvent();
    event.action = action;
    return event;
  }

  @Action public int getAction() {
    return action;
  }

  @IntDef({
      CONNECT,
      DISCONNECT
  })
  @Retention(RetentionPolicy.SOURCE)
  public @interface Action {

  }
}

File: remote/src/main/kotlin/com/kelsos/mbrc/events/ChangeWebSocketStatusEvent.kt
package com.kelsos.mbrc.events

import android.support.annotation.IntDef

class ChangeWebSocketStatusEvent private constructor()//no instance
{

  @Action var action: Int = 0
    private set

  @IntDef(CONNECT , DISCONNECT)
  @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
  annotation class Action

  companion object {
    const val CONNECT = 1L
    const val DISCONNECT = 2L

    fun newInstance(@Action action: Int): ChangeWebSocketStatusEvent {
      val event = ChangeWebSocketStatusEvent()
      event.action = action
      return event
    }
  }
}
